### PR TITLE
User repository fetch with access rights

### PR DESCRIPTION
**Pull request by taher435**
Issue: https://github.com/indydevs/parma/issues/2

### Things to do
- [x] Fix all compiler warnings
- [x] Fetch user repositories along with all the access details
- [x] Add bootstrap sass and configure brunch to work with sass
- [x] Beautify user repository list page (with bootstrap)
- [x] Add on/off toggle to enable/disable webhook to repository (setting in db)
- [x] When `on` add a `webhook` to the github repo to call to a placeholder webhook URL in parma
- [x] When `off` remove the `webhook`
- [x] Add specs

### Screenshots

image

image

##### Hook Added

image

### CODE REVIEW

File: web/models/user_from_auth.ex [hunk 1]
```diff
@@ -94,24 +94,28 @@ defmodule Parma.UserFromAuth do
...
+      User.registration_changeset(%User{}, scrub(%{email: auth.info.email, name: name}))
+      |> repo.insert
     case result do
-      {:ok, user} -> user
-      {:error, reason} -> repo.rollback(reason)
+      {:ok, user} -> 
+        Toniq.enqueue(UserInitialization, user: user)
+        user
+      {:error, reason} -> 
+        repo.rollback(reason)
```
Comment (ankitagupta12) [inline web/models/user_from_auth.ex:118]: so we have to manually rollback?

File: web/models/user_from_auth.ex [hunk 2]
```diff
@@ -183,7 +187,7 @@ defmodule Parma.UserFromAuth do
       auth.info.name
     else
       [auth.info.first_name, auth.info.last_name]
-      |> Enum.filter(&(&1 != nil and String.strip(&1) != ""))
+      |> Enum.filter(&(&1 != nil and String.trim(&1) != ""))
```
Comment (ankitagupta12) [inline web/models/user_from_auth.ex:190]: is there a reason why you prefer `trim` over `strip`

File: web/services/repository_service.ex [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+  def enable(%Repository{} = repository, %User{} = user) do
+    owner = get_repo_owner(repository)
+
+    {_status, response} = create_hook(owner, repository.name, user)
+
+    Repository.update(repository, %{enabled: true, hook_id: response|> Map.get("id") |> to_string})
+  end
+
+  def enable(_, _), do: not_found_error()
```
Comment (ankitagupta12) [inline web/services/repository_service.ex:17]: is this exercised when the user or repository is nil?

File: web/models/user_from_auth.ex (hunk 2 above)
Comment (taher435) [reply web/models/user_from_auth.ex:190]: `strip` is deprecated
https://github.com/elixir-lang/elixir/blob/v1.3/CHANGELOG.md#3-soft-deprecations-no-warnings-emitted

File: web/models/user_from_auth.ex (hunk 1 above)
Comment (taher435) [reply web/models/user_from_auth.ex:118]: Yes, unless in an Ecto transaction or Multi, we need to rollback manually.

File: web/services/repository_service.ex (hunk 3 above)
Comment (taher435) [reply web/services/repository_service.ex:17]: Yes, this helps avoid a `nil` check inside the function.

**Review by ankitagupta12 — APPROVED** [review]

**Pull request merged**